### PR TITLE
Multiple application ID can be set on android platform

### PR DESCRIPTION
**Pull request by Md-Abdul-Halim-Rafi**

### CODE REVIEW

File: android/src/main/java/ru/innim/flutter_login_facebook/MethodCallHandler.java [hunk 1]
```diff
@@ -30,8 +30,11 @@
     private final static String _GET_SDK_VERSION = "getSdkVersion";
     private final static String _GET_USER_EMAIL = "getUserEmail";
     private final static String _GET_PROFILE_IMAGE_URL = "getProfileImageUrl";
+    private final static String _SET_APP_ID = "setAppId";
 
     private final static String _PERMISSIONS_ARG = "permissions";
+    private final static String _APP_ID_ARG = "appIdArg";
```
Comment (greymag) [inline android/src/main/java/ru/innim/flutter_login_facebook/MethodCallHandler.java:36]: Name of argument (value string) should be `appId`, without `Arg`

File: example/lib/main.dart [hunk 2]
```diff
@@ -118,7 +118,7 @@ class _MyHomeState extends State<MyHome> {
   Future<void> _onPressedLogInButton() async {
     await widget.plugin.logIn(permissions: [
       FacebookPermission.publicProfile,
-      FacebookPermission.email,
+      FacebookPermission.email
```
Comment (greymag) [inline example/lib/main.dart:121]: Do not remove this trailing comma

File: ios/Classes/SwiftFlutterLoginFacebookPlugin.swift [hunk 3]
```diff
@@ -49,6 +54,18 @@ public class SwiftFlutterLoginFacebookPlugin: NSObject, FlutterPlugin {
             
             let permissions = permissionsArg.map {val in Permission(stringLiteral: val)}
             logIn(result: result, permissions: permissions)
+        case .setAppId:
+            guard
+                let args = call.arguments as? String: Any,
+                let appId = args[AppIdArg.permissions.rawValue] as? String
```
Comment (greymag) [inline ios/Classes/SwiftFlutterLoginFacebookPlugin.swift:60]:
```suggestion
                let appId = args[AppIdArg.appId.rawValue] as? String
```

File: ios/Classes/SwiftFlutterLoginFacebookPlugin.swift [hunk 4]
```diff
@@ -234,6 +251,30 @@ public class SwiftFlutterLoginFacebookPlugin: NSObject, FlutterPlugin {
             result(data)
         }
     }
+
+    private func setAppId(result: @escaping FlutterResult, appId: AppId) {
+        let viewController = (UIApplication.shared.delegate?.window??.rootViewController)!
+        
+        _loginManager.setAppId(
```
Comment (greymag) [inline ios/Classes/SwiftFlutterLoginFacebookPlugin.swift:258]: I can't see such method in `LoginManager`, but I can see `setAppId` in `FBSDKSettings` (`Settings` in Swift)
https://developers.facebook.com/docs/reference/ios/current/constants/FBSDKSettings/

File: ios/Classes/SwiftFlutterLoginFacebookPlugin.swift [hunk 5]
```diff
@@ -234,6 +251,30 @@ public class SwiftFlutterLoginFacebookPlugin: NSObject, FlutterPlugin {
...
+            FBSDKSettings.setDefaultAppID(appId)
+        
+            let data: [String: Any?] = [
+                "status": status,
+                "accessToken": accessTokenMap,
+                "error": errorMap
+            ]
+            
+            result(data)
+        }
```
Comment (greymag) [inline ios/Classes/SwiftFlutterLoginFacebookPlugin.swift:276]: It shouldn't be here

File: ios/flutter_login_facebook.podspec [hunk 6]
```diff
@@ -16,6 +16,7 @@ Login via Facebook for Flutter projects.
   s.source_files = 'Classes/**/*'
   s.dependency 'Flutter'
   s.dependency 'FBSDKLoginKit', '~> 12.0'
+  s.dependency 'FBSDKCoreKit', '~> 12.0'
```
Comment (greymag) [inline ios/flutter_login_facebook.podspec:19]: Do we really need this?

File: lib/src/facebook_login.dart [hunk 7]
```diff
@@ -16,8 +16,10 @@ class FacebookLogin {
   static const _methodGetUserEmail = 'getUserEmail';
   static const _methodGetProfileImageUrl = 'getProfileImageUrl';
   static const _methodGetSdkVersion = 'getSdkVersion';
+	static const _methodSetAppId = 'setAppId';
```
Comment (greymag) [inline lib/src/facebook_login.dart:19]: Check your IDE settings - indents are different

File: lib/src/facebook_login.dart [hunk 8]
```diff
@@ -16,8 +16,10 @@ class FacebookLogin {
...
 
   static const _permissionsArg = 'permissions';
+	static const _appIdArg = 'appIdArg';
```
Comment (greymag) [inline lib/src/facebook_login.dart:22]: Argument name `appId`

File: lib/src/facebook_login.dart [hunk 9]
```diff
@@ -26,9 +28,13 @@ class FacebookLogin {
 
   /// If `true` all requests and results will be printed in console.
   final bool debug;
+	/// Initialize with Facebook App ID
+  final String appId;
 
-  FacebookLogin({this.debug = false}) {
+  FacebookLogin({this.debug = false, required this.appId}) {
```
Comment (greymag) [inline lib/src/facebook_login.dart:34]: Shouldn't be required. It's completely optional and all existed projects should be working without any changes.

File: lib/src/facebook_login.dart [hunk 10]
```diff
@@ -50,6 +56,17 @@ class FacebookLogin {
     return _isLoggedIn(token);
   }
 
+	/// Set Application ID.
+	Future<void> setAppId({String? appId}) {
```
Comment (greymag) [inline lib/src/facebook_login.dart:60]: Make `appId` non nullable and required. In this implementation there is no sense to call this method without an `appId`.

File: android/build.gradle [hunk 11]
```diff
@@ -35,4 +35,5 @@ android {
 
 dependencies {
     implementation 'com.facebook.android:facebook-login:12.0.1'
+    implementation 'com.facebook.android:facebook-android-sdk:5.+'
```
[the remark this answers is not in the document]
Comment (Md-Abdul-Halim-Rafi) [reply android/build.gradle:38]: But, only login SDK doesn't allow to set application ID dynamically as far as I know.

[the remark this answers is not in the document]
Comment (greymag) [reply android/build.gradle:38]: So you should find out what exactly we need to allow that. For sure we don't need whole SDK.

**Comment by greymag** [comment]
No changes, so I'm closing it for now. Feel free to reopen PR if all review notes would be implemented.

**Pull request closed without merging**